syntax = "proto3";

import "protocol/pair.proto";
import "protocol/device_sharing.proto";
import "protocol/clipboard.proto";
import "protocol/fs.proto";
import "protocol/service_status.proto";

message Message {
  oneof payload {
    ScanRequest scanRequest = 1000;
    ScanResponse scanResponse = 1001;
    PairRequest pairRequest = 1002;
    PairResponse pairResponse = 1003;
    StopPairRequest stopPairRequest = 1004;
    StopPairResponse stopPairResponse = 1005;
    ServiceStoppedNotification serviceStoppedNotification = 1100;
    ServiceOnOffNotification serviceOnOffNotification = 1200;

    DeviceSharingStartRequest deviceSharingStartRequest = 2000;
    DeviceSharingStartResponse deviceSharingStartResponse = 2001;
    DeviceSharingStopRequest deviceSharingStopRequest = 2002;
    DeviceSharingStopResponse deviceSharingStopResponse = 2003;
    FlowRequest flowRequest = 2100;
    FlowResponse flowResponse = 2101;

    FsRequest fsRequest = 3000;
    FsResponse fsResponse = 3001;
    FsSendFileRequest fsSendFileRequest = 3002;
    FsSendFileResponse fsSendFileResponse = 3003;
    FsSendFileResult fsSendFileResult = 3004;
    FsMethodGetAttrRequest fsMethodGetAttrRequest = 3100;
    FsMethodGetAttrResponse fsMethodGetAttrResponse = 3101;
    FsMethodReadRequest fsMethodReadRequest = 3102;
    FsMethodReadResponse fsMethodReadResponse = 3103;
    FsMethodReadDirRequest fsMethodReadDirRequest = 3104;
    FsMethodReadDirResponse fsMethodReadDirResponse = 3105;
    FsMethodOpenRequest fsMethodOpenRequest = 3106;
    FsMethodOpenResponse fsMethodOpenResponse = 3107;
    FsMethodReleaseRequest fsMethodReleaseRequest = 3108;
    FsMethodReleaseResponse fsMethodReleaseResponse = 3109;

    InputEventRequest inputEventRequest = 4000;
    InputEventResponse inputEventResponse = 4001;

    ClipboardNotify clipboardNotify = 5000;
    ClipboardGetContentRequest clipboardGetContentRequest = 5001;
    ClipboardGetContentResponse clipboardGetContentResponse = 5002;

    // RemoteDisplayConfigRequest remoteDisplayConfigRequest = 6000;
    // RemoteDisplayConfigResponse remoteDisplayConfigResponse = 6001;
    // RemoteDisplaySendRequest remoteDisplaySendRequest = 6002;
    // RemoteDisplaySendResponse remoteDisplaySendResponse = 6003;
  }
}
